pkg/controllers/member/serviceexport/controller.go

```diff
@@ -180,9 +181,24 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 	}
 
+	// Get the weight from the serviceExport annotation and validate it.
+	exportWeight, err := extractWeightFromServiceExport(&svcExport)
```

Comment:
should we validate the annotation before we add the finalizer?

if the finalizer is added, need to unexportService (delete internalServiceExport in hub).

You can refer to line 155 about how we handle isServiceEligibleForExport,

Reply:
btw, add event to be consistent with the existing code,

for example,

r.Recorder.Eventf(&svcExport, corev1.EventTypeWarning, "ServiceNotFound", "Service %s is not found or in the deleting state", svc.Name)

Reply:
could you please add a comment to call this out? considering the weight is specified as an annotation, not updating internalServiceExport is better since it's hard to block the invalid changes compared with a field in the spec,